go-mode.el

```diff
@@ -2020,6 +2034,10 @@ If BUFFER, return the number of characters in that buffer instead."
   (with-current-buffer (or buffer (current-buffer))
     (1- (position-bytes (point-max)))))
 
+(defun go--current-workspace ()
```

Comment:
With go modules I think the name of this function is confusing. If you are outside of the GOPATH your workspace isn't your GOPATH.

I'm also not sure when you would want to match GOROOT. Are you commonly editing files in your GOROOT?